File: subprojects/frida-node/examples/snapshot.ts
import frida from "frida";

const embedScript = `
const button = {
  color: 'blue',
};

function mutateButton() {
  button.color = 'red';
}
`;

const warmupScript = `
mutateButton();
`;

const testScript = `
console.log('Button before:', JSON.stringify(button));
mutateButton();
console.log('Button after:', JSON.stringify(button));
`;

const runtime = frida.ScriptRuntime.V8;

async function main() {
    const session = await frida.attach(0);

    const snapshot = await session.snapshotScript(embedScript, { warmupScript, runtime });

    const script = await session.createScript(testScript, { snapshot, runtime });
    script.message.connect(message => {
        console.log("[*] Message:", message);
    });
    await script.load();

    await script.unload();
}

main()
    .catch(e => {
        console.error(e);
    });

File: subprojects/frida-python/examples/snapshot.py
import frida

embed_script = """
const button = {
  color: 'blue',
};

function mutateButton() {
  button.color = 'red';
}
"""

warmup_script = """
mutateButton();
"""

test_script = """
console.log('Button before:', JSON.stringify(button));
mutateButton();
console.log('Button after:', JSON.stringify(button));
"""

runtime = "v8"


session = frida.attach(0)

snapshot = session.snapshot_script(embed_script, warmup_script=warmup_script, runtime=runtime)


def on_message(message, data):
    print("on_message:", message)


script = session.create_script(test_script, snapshot=snapshot, runtime=runtime)
script.on("message", on_message)
script.load()
